From 9c6de4758153a305b8357f8ee4b811adb49c3c4a Mon Sep 17 00:00:00 2001
From: Weiguo Hu <hwg@rock-chips.com>
Date: Wed, 4 Nov 2020 15:03:04 +0800
Subject: [PATCH] net: rockchip_wlan: rtl8822bs: solve WARNING on
 nl80211_send_chandef

WARNING: CPU: 3 PID: 516 at net/wireless/nl80211.c:2880 nl80211_send_chandef+0x16c/0x180
Call trace:
  nl80211_send_chandef
  nl80211_ch_switch_notify
  cfg80211_ch_switch_started_notify
  rtw_cfg80211_ch_switch_notify
  rtw_chk_start_clnt_join
  join_cmd_hdl
  rtw_cmd_thread
  kthread

Signed-off-by: Weiguo Hu <hwg@rock-chips.com>
Change-Id: I922c8bd3af5de802dd7f2ee46738ff3fe07ca2af
---
 .../rtl8822bs/os_dep/linux/ioctl_cfg80211.c              | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rtl8822bs/os_dep/linux/ioctl_cfg80211.c b/drivers/net/wireless/rockchip_wlan/rtl8822bs/os_dep/linux/ioctl_cfg80211.c
index d84339908659..8c5c4a4a2a06 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8822bs/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8822bs/os_dep/linux/ioctl_cfg80211.c
@@ -202,6 +202,8 @@ static u8 rtw_chbw_to_cfg80211_chan_def(struct wiphy *wiphy, struct cfg80211_cha
 	struct ieee80211_channel *chan;
 	u8 ret = _FAIL;
 
+	_rtw_memset(chdef, 0, sizeof(*chdef));
+
 	freq = rtw_ch2freq(ch);
 	if (!freq)
 		goto exit;
@@ -225,6 +227,12 @@ static u8 rtw_chbw_to_cfg80211_chan_def(struct wiphy *wiphy, struct cfg80211_cha
 		chdef->width = NL80211_CHAN_WIDTH_80;
 	else if (bw == CHANNEL_WIDTH_160)
 		chdef->width = NL80211_CHAN_WIDTH_160;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0))
+	else if (bw == CHANNEL_WIDTH_5)
+		chdef->width = NL80211_CHAN_WIDTH_5;
+	else if (bw == CHANNEL_WIDTH_10)
+		chdef->width = NL80211_CHAN_WIDTH_10;
+#endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)) */
 	else {
 		rtw_warn_on(1);
 		goto exit;
@@ -232,7 +240,6 @@ static u8 rtw_chbw_to_cfg80211_chan_def(struct wiphy *wiphy, struct cfg80211_cha
 
 	chdef->chan = chan;
 	chdef->center_freq1 = cfreq;
-	chdef->center_freq2 = 0;
 
 	ret = _SUCCESS;
 
-- 
2.35.3

